src/main/java/com/urswolfer/gerrit/client/rest/http/projects/ProjectApiRestClient.java

```diff
@@ -92,6 +97,27 @@ public BranchApi branch(String ref) throws RestApiException {
         JsonElement branches = gerritRestClient.getRequest(request);
         return branchInfoParser.parseBranchInfos(branches);
     }
+    
+    @Override
+    public ListRefsRequest<TagInfo> tags() {
+        return new ListRefsRequest<TagInfo>() {
+            @Override
+            public List<TagInfo> get() throws RestApiException {
+                return ProjectApiRestClient.this.getTagInfos(this);
+            }
+        };
+    }
+    
+    @Override
+    public TagApi tag(String ref) throws RestApiException {
+        return new TagApiRestClient(gerritRestClient, tagInfoParser, this, ref);
+    }
+    
+    private List<TagInfo> getTagInfos(ListRefsRequest<TagInfo> lbr) throws RestApiException {
```

Comment:
Yes, you were right.
I implemented it as you had described.
Please have a look.